well_known.go

```diff
@@ -12,11 +12,12 @@ import (
 // Located at https://<server_name>/.well-known/matrix/server
 type WellKnownResult struct {
 	NewAddress ServerName `json:"m.server"`
+	Error      string     `json:"Error,omitempty"`
```

Comment:
this doesn't feel like the right thing to be doing. `WellKnownResult` is supposed to represent the thing that gets returned from the server. Surely we shouldn't be adding extra fields to it? Make another structure if need be.